schema/configuration.go

```diff
@@ -55,7 +55,15 @@ type DataSharing struct {
 	PushTo              []string
 }
 
-var MalformedConfigError = errors.New("config file is malformed")
+type MalformedConfigError struct {
+	What string
```

Comment:
Better yet, I see you make MalformedConfigErrors in four different ways.

- MalformedConfigError{What: "JSON error"} (general error case)
- MalformedConfigError{What: sectionError(JSONAPI)} (specific section case)
- MalformedConfigError{What: keyError(JSONAPI, "HTTPHeaders")} (key within section case)
- MalformedConfigError{What: keyError("", "Dropbox-api-token")} (key outside section case?)

Instead of making it the responsibility of the dev using `MalformedConfigError` to understand how to use each of the cases, you should make helpers which produce the right version of `MalformedConfigError`. Something like:

- malformedConfigError("invalid raw JSON") // we already know it's a error, don't repeat yourself...
- malformedConfigError("invalid section JSONAPI")
- malformedConfigError("JSONAPI", "HTTPHeaders") // section-specific helper function
- malformedConfigError("Dropbox-api-token") // key-specific helper function outside of a section

Instead consider variadic parameters:
```
func malformedConfigKey(pathArgs ...string) malformedConfigError {
  return malformedConfigKey{path: pathArgs...}
}

func (err malformedConfigKey) Error() string {
  return fmt.Sprintf("malformed config: %s", strings.Join(err.path..., "."))
}
```

This makes `malformedConfigError` have a "variadic" parameter which is treated like a slice. Then you can pass in as many values as you like to describe how "deep" the malformed data is in the JSON.

Ex: 
```
malformedConfigError("JSON is invalid") // "malformed config: JSON is invalid"
malformedConfigError("JSON-API","config") // "malformed config: JSON-API.config"
malformedConfigError("JSON-API","space1","value1") // "malformed config: JSON-API.space1.value1"
```